pkg/kubelet/kubelet.go

```diff
@@ -1471,37 +1471,31 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	updateType := o.updateType
 
 	//Get the hostname for traces
-	activeHost, err := os.Hostname()
-	if err != nil {
-		return fmt.Errorf("could not retrieve hostname")
-	}
+	activeHost := string(kl.nodeName)
 
 	// Create and register an OpenCensus
 	// Stackdriver Trace exporter.
 	exporter, err := stackdriver.NewExporter(stackdriver.Options{
 		ProjectID: "samnaser-gke-dev-217421",
 	})
-
 	if err != nil {
 		return fmt.Errorf("error configuring stackdriver opencensus exporter: %v", err)
 	}
 
 	// Register our exporter with our tracer and set a liberal sampling policy for testing
 	trace.RegisterExporter(exporter)
-	trace.ApplyConfig(trace.Config{DefaultSampler: trace.AlwaysSample()})
+	trace.ApplyConfig(trace.Config{DefaultSampler: trace.ProbabilitySampler(0.15)})
 
 	// Parent span, annotate it with the hostname
 	ctx := context.Background()
 	ctx, span := trace.StartSpan(ctx, "Kubelet: SyncPod")
 	span.AddAttributes(trace.StringAttribute("host", activeHost))
 	defer span.End()
 
-	podKillCtx, podKillSpan := trace.StartSpan(ctx, "Pod kill logic")
-
 	// if we want to kill a pod, do it now!
 	if updateType == kubetypes.SyncPodKill {
 
-		podKillSpan.Annotate([]trace.Attribute{}, "Pod being killed")
+		_, podKillSpan := trace.StartSpan(ctx, "Pod kill logic")
```

Comment:
does this still work if we don't overwrite the `ctx`?  Previously, you passed the context returned from the `StartSpan` for kill pod to the next StartSpan call.

Reply:
In this case, since we `return` no matter what at the end of this block, I believe we can just use the existing context to begin the span and don't need to create a new `ctx`

Reply:
ah, ok.  Sounds good